### PR TITLE
Insulin Delivery

### PR DESCRIPTION
**Pull request by bassrock**
This PR adds Insulin Delivery to the Application and also uploads it to Apple Health and Nightscout.

I also updated the Health integration to delete glucose and insulin records from Apple Health when they originated from Glucose Direct and they are deleted within Glucose Direct.

I chose to use the insulin options that NightScout uses (Meal Bolus, Correction Bolus, Snack Bolus) and then I downgrade them for use in Apple Health which only supports Bolus and Basal.

Some of these requests are in #214

### CODE REVIEW

File: App/Views/LogInsulinView.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    
+    @FocusState private var unitsFocus: Bool
+    
+    var addCallback: ((_ starts: Date, _ ends: Date, _ units: Float, _ insulinType: InsulinType) -> Void)
+    
+    var body: some View {
+        NavigationView {
+            HStack {
+                Form {
+                    Section(footer: Text("Basal insulin referes to the insulin used to regulate blood glucose between meals including during sleep. Bolus insulin refers to the insulin used to regulate blood glucose at meals and/or to acutely address high blood glucose.")) {
```
Comment (bassrock) [inline App/Views/LogInsulinView.swift:25]: Copy that is similar to Apple Health data. Could remove if wanted.

**Comment by creepymonster** [comment]
I really like the sheet for entering the insulin. I will definitely adopt this for entering blood glucose values/calibrations.

**Comment by bassrock** [comment]
@creepymonster thanks! Are there any steps you'd like me to take to get this good for merging?

**Comment by creepymonster** [comment]
No, I think it's great that way. I will do the merge and in the next step take over the sheet for glucose values/calibrations.

A topic that will then be interesting is the display of insulin in the chart (but no longer the compatibility chart for iOS 15 devices).

**Comment by creepymonster** [comment]
Thank you very much, I was very excited about the PR yesterday.

I myself have always shied away from it, because I myself do not enter carbohydrates ☺️. I have always entered them in FAX/Loop.

**Comment by bassrock** [comment]
@creepymonster yep! I was thinking about displaying it on the chart as well in a future pr, similar to how the libre app does it.

**Pull request merged**